File: csharp/DesiArchitect.DesignPatterns/07_Facade/After/SeatService.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Facade.After;

public class SeatService : ISeatService
{
    public bool Reserve(BookingRequest request)
    {
        Console.WriteLine($"  🪑 SeatService: Reserved seat on train {request.TrainNumber}");
        return true;
    }
}

File: typescript/src/07-facade/after/seat-service.ts
import type { BookingRequest } from '../../shared/models.js';
import type { ISeatService } from './i-seat-service.js';

export class SeatService implements ISeatService {
  reserve(request: BookingRequest): boolean {
    console.log(`  🪑 SeatService: Reserved seat on train ${request.trainNumber}`);
    return true;
  }
}